pandas/core/arrays/categorical.py

```diff
@@ -1229,11 +1229,26 @@ def map(self, mapper):
         Index(['first', 'second', nan], dtype='object')
         """
         new_categories = self.categories.map(mapper)
+
         try:
-            return self.from_codes(self._codes.copy(),
-                                   categories=new_categories,
-                                   ordered=self.ordered)
+            if isinstance(mapper, (dict, ABCSeries)):
+                new_value = mapper[np.nan]
+            else:
+                new_value = mapper(np.nan)
+        except (AttributeError, KeyError, TypeError, ValueError):
+            new_value = np.nan
+
+        try:
```

Comment:
why are you try/except here?

Reply:
how / why can this fail

Reply:
AttributeError: if mapper calls a method of the element (e.g. `lambda x: x.lower()`)
KeyError: if mapper is a dict without a key for NaN
TypeError: if mapper expects some type other than a float
ValueError: if mapper tries converting float values to ints (e.g. `lambda x: int(x)`)